amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -1,743 +1,322 @@
-
-    return test_func
-
-
-# Benchmarks pure python event based write API
-# https://github.com/amazon-ion/ion-python/issues/236
-def write_micro_benchmark_event(iterations, warmups, c_extension, file, binary, memory_profiling, io_type,
-                                format_option):
-    return 0, 0
-
-
-# Framework for benchmarking write methods, this functions includes
-# 1. profile memory usage,
-# 2. benchmark performance,
-# 3. generate report
-def write_micro_benchmark_and_profiling(table, write_micro_benchmark_function, iterations, warmups, file, c_extension,
-                                        binary, each_option, io_type, command):
-    if not write_micro_benchmark_function:
-        raise Exception("Invalid micro benchmark function: micro benchmark function can not be none.")
-    # Memory Profiling
-    if not pypy:
-        write_micro_benchmark_function(iterations=1, warmups=0, c_extension=c_extension, file=file,
-                                       binary=binary, memory_profiling=True, format_option=each_option[1],
-                                       io_type=io_type)
-
-    # Performance Benchmark
-    file_size, result_with_gc = \
-        write_micro_benchmark_function(iterations=iterations, warmups=warmups, c_extension=c_extension,
-                                       file=file, binary=binary, memory_profiling=False, format_option=each_option[1],
-                                       io_type=io_type)
-
-    # generate report
-    write_generate_report(table, file, file_size, command, each_option, result_with_gc, write_memory_usage_peak)
-
-    return file_size, each_option, result_with_gc, write_memory_usage_peak
-
-
-# Generates and prints benchmark report
-def write_generate_report(table, file, file_size, command, each_option, total_time, memory_usage_peak):
-    insert_into_report_table(table,
-                             [file,
-                              format_decimal(file_size),
-                              command,
-                              each_option,
-                              format_decimal(total_time),
-                              format_decimal(memory_usage_peak)])
-
-
-# Insert a benchmark result row into the benchmark report (table)
-def insert_into_report_table(table, row):
-    if not isinstance(row, list):
-        raise Exception('row must be a list')
-    table += [row]
-
-
-# Create a report table by given description
-def identify_report_table(command):
-    if command == 'read':
-        return identify_report_table_helper(
-            ['file_name', 'file_size (MB)', 'command', 'options', 'total_time (s)', 'memory_usage_peak (MB)'])
-    elif command == 'write':
-        return identify_report_table_helper(
-            ['file_name', 'file_size (MB)', 'command', 'options', 'total_time (s)', 'memory_usage_peak (MB)']
-        )
-    else:
-        raise Exception('Command should be either read or write.')
+    iterations = int(arguments['--iterations'])
```

Comment:
Yea at least with `argparse` this is all done for you. I like `docopt` at a high-level and I don't recommend changing, just an FYI.

Reply:
I tried it, and `docopt` doesn't convert strings to other types for you. Then I found this article that seems to confirm it: https://click.palletsprojects.com/en/3.x/why/#why-not-docopt-etc